### PR TITLE
Add .travis.yml

### PR DESCRIPTION
**Pull request by mezz**
Testing out Travis CI for Forge PRs

### CODE REVIEW

**Comment by liach** [comment]
Should it run `./gradlew installer`?

**Comment by liach** [comment]
I suggest you to use java 6 and 8 together.

Edit: it seems forgegradle requires java7 to run.

**Comment by mezz** [comment]
I am taking it one step at a time.

File: .travis.yml [hunk 1]
```diff
@@ -20,9 +20,10 @@ cache:
 before_install: chmod +x gradlew
 install:
   - ./gradlew --refresh-dependencies
+  - ./gradlew ciWriteBuildNumber --stacktrace
 script:
-  - ./gradlew ciWriteBuildNumber
-  - ./gradlew Forge:test
+  - ./gradlew build --stacktrace
+  - ./gradlew Forge:test --stacktrace
```
Comment (liach) [inline .travis.yml]: ./gradlew test -S?

Comment (mezz) [reply .travis.yml]: probably yeah

**Review by liach — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    - $HOME/.gradle/wrapper/
+
+before_install: chmod +x gradlew
+install:
+  - ./gradlew --refresh-dependencies
+  - ./gradlew ciWriteBuildNumber
+before_script: wget -O /tmp/init.gradle https://gist.githubusercontent.com/cpw/fe8a6ee27bde84d8731a/raw/23341d9838a99eddb323b87a32d03c4e6fd57526/bootstrap.gradle
+script:
+  - ./gradlew -I /tmp/init.gradle build -S
+  - ./gradlew test -S
```
Comment (liach) [inline .travis.yml]: It seems that there is no gradle task to run tests.

Comment (mezz) [reply .travis.yml]: Yeah, I'm not sure how to run them and compile the Forge test mods.

**Comment by mezz** [comment]
Trying to get Java 6 working has been very difficult, I'm giving up on that part for now.
Everything else seems to be working well so this is looking useful now.

**Comment by AEnterprise** [comment]
@mezz you should be able to force travis to use java 6 by switching to it
https://docs.travis-ci.com/user/languages/java/#Switching-JDKs-Within-One-Job

**Comment by mezz** [comment]
ForgeGradle needs Java 7 or higher for some parts (like downloading files over https), and it's not simple to break up the build task into parts as far as I can tell. Let me know if you have an idea to try.

**Comment by AEnterprise** [comment]
are any of those parts done every time? i don't know the exact internals but once it downloaded the files they don't need to be downloaded again (i presume), so if everything from java 7 is cached then it could be a simple as running it twice, once uncer 7 and then under 6 to verify compatibility

**Comment by mezz** [comment]
I think there are other parts of the process that won't run under Java 6, that's just the one I ran into first.

**Comment by AEnterprise** [comment]
alright i don't know the exact internals of the build task code so i could be wrong here. But would it be an option to split the task into 2 seperate tasks? the first being what needs java 7 and a second that can run with 6 (where the actual compiling heapens), if the current build task then is replaced to just call those 2 new subtasks things keep working as  they are but both parts can be called seperatly with other jdk's by travis

**Comment by matthewprenger** [comment]
Minecraft itself doesn't even compile on Java 6 anymore, and hasn't for a while. Runtime is still Java 6 compatible but development needs 7 or higher because of certain generics used.

**Comment by liach** [comment]
https://github.com/SpongePowered/SpongeCommon/blob/1b16d435e7d056c28364253e7edc6611cf762b43/gradle/minecraft.gradle#L46-L49

Adding source and target compatibility probably helps.

**Comment by LexManos** [comment]
Not true, MCP works just fine on J6 according to all my tests. There was a break a while back due to a misshap i had in the patches, but that was quickly fixed. Please stop spreading lies.
Again, The point isn't what we compile for its what we RUN for.
Sadly the compiler does not process @seince tags when compiling so if you compile on the J7 JDK you can have issues when targeting J6. The bytecode will be valid, but you can happily reference classes/functions that did not exist. Which is what they are trying to solve.

**Comment by mezz** [comment]
I will try to get Java 6 working later, I need a break from gradle.
For now, this is a useful stopping point and is ready to be pulled.

**Pull request merged**